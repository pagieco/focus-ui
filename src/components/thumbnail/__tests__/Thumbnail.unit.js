import { shallowMount } from '@vue/test-utils';
import { Thumbnail } from '../index';

const shallowWrap = propsData => shallowMount(Thumbnail, { propsData });

describe('Thumbnail', () => {
  it('is a valid vue component', () => {
    expect(Thumbnail).toBeAComponent();
  });

  it('matches the snapshot', () => {
    const wrapper = shallowWrap({ alt: 'My Alt Text', source: 'image.png' });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('uses the alt prop', () => {
    const wrapper = shallowWrap({ alt: 'My Alt Text', source: 'image.png' });

    expect(wrapper.find('img').attributes('alt')).toBe('My Alt Text');
  });

  it('uses the size prop', () => {
    const wrapper = shallowWrap({
      alt: 'My Alt Text',
      source: 'image.png',
      size: 'small',
    });

    expect(wrapper.classes()).toContain('thumbnail--size-small');
  });

  it('uses the source prop', () => {
    const wrapper = shallowWrap({ alt: 'My Alt Text', source: 'image.png' });

    expect(wrapper.find('img').attributes('src')).toBe('image.png');
  });
});
