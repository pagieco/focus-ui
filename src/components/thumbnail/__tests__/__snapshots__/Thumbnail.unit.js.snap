// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnail matches the snapshot 1`] = `
<span
  class="thumbnail thumbnail--size-medium"
>
  <img
    alt="My Alt Text"
    src="image.png"
  />
</span>
`;
